bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -501,11 +507,39 @@ public Widget getWidget(Sitemap sitemap, String id) {
      */
     @Override
     public EList<Widget> getChildren(LinkableWidget w) {
+        EList<Widget> widgets = null;
         if (w instanceof Group && w.getChildren().isEmpty()) {
-            return getDynamicGroupChildren((Group) w);
+            widgets = getDynamicGroupChildren((Group) w);
         } else {
-            return w.getChildren();
+            widgets = w.getChildren();
         }
+
+        EList<Widget> result = new BasicEList<Widget>();
+        for (Widget widget : widgets) {
+            Widget resolvedWidget = resolveDefault(widget);
+            if (resolvedWidget != null) {
+                result.add(resolvedWidget);
+            } else {
+                result.add(widget);
+            }
+        }
+        return result;
+    }
+
+    private Widget resolveDefault(Widget widget) {
```

Comment:
Wouldn't the code be simpler (less <code>if</code>s above) if we
* <code>return widget;</code> if it is not an instance of Default
* change the method so you can give the fallback: <code>private Widget resolveDefault(Widget widget, Widget fallback)</code> and <code>return fallback;</code> at the end (so you can use resolveDefault(widget, null) or resolveDefault(widget, widget)